### PR TITLE
Plumbing (Indefinitely postponed)

### PR DESCRIPTION
**Pull request by ghost**
Nanotrasen has investigated the magical amounts of infinite water appearing from various appliances and discovered a corruption of the Wizard's Federation! Thankfully they managed to get rid of it, but as a result they've begun installing plumbing appliances and hired a few ~~bumbling idiots~~ trained professionals to manage the station's water supply.
### The Way it Works

The **Plumber** has been added. By default, there are two Plumbers. Their job is to manage the station's water supply and ensure that it remains untainted and stable. The main way to do this is via manual control of the various machinery.

The **Central Plumbing Synthesizer** will automatically generate a single type of reagent and has an interior storage of 100,000 units. By default, this is water, but by doing complex maintenance procedures one can access the scanner and change the reagent it will synthesize. This means you can use it for nefarious purposes, like making the sinks spit out acid, or helpful ones, like using healing chemicals if there's a crisis! Unfortunately, it's very single-minded, and will not allow you to control the spread of reagents to specific systems or areas; the reagents will simply be distributed to everything.

Plumbers are the only ones who can understand all the complex workings of the machine by default, but there are a few ways to let other crew members understand them. Using the main control terminal, Plumbers can change the speed at which reagents synthesize, the pressure at which they are sent to the rest of the station, flush the tank of its reagents, and much more.

Due to cutting-edge research, all objects that require reagents are automatically fed by miniature pipes embedded in the station's flooring itself. This means that you will not have to lay down piping from place to place.

Replacing the reagents is possible. The maintenance procedure, for now, is as follows:
1. Access the administrator rights of the terminal.
2. Allow maintenance protocols on the synthesizer.
3. Move around to the other side of the machine and open the maintenance panel.
4. Disconnect the synthesizer from the rest of the station by pulling a lever.
5. Return to the terminal and disable synthesization.
6. Disable the lock on the cover of the synthesizer.
7. Use a crowbar and pull back the cover.

From here, you can do a few things, including hiding things in the synthesizer's tank, throwing people into it, diving into it, pouring reagents in, etc. Alternatively:
1. Access the scanner terminal.
2. Scan a reagent you wish to synthesize with a container holding some of it.

To disable maintenance, simply do the steps in reverse.

But what good would plumbing be without more things to use it? A few new objects will be added for this, including:

**Sprinklers**, which will trigger automatically in the event of a fire alarm and spray ~~chlorine trifluoride~~ water all over the area!

**Heavy-Duty Faucets** which can rapidly fill up a container with reagents!

**Water Fountains** which can dispense revitalizing water or debilitating poisons!

In addition, sinks, toilets, and showers now use plumbing.
### Coming soon to a station near you!

_(I will likely be working on this once the rest of my PRs are merged.)_

### CODE REVIEW

**Comment by Nienhaus** [comment]
10/10 Plumbing dmi.

File: code/game/machinery/plumbing/synthesizer.dm [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		user << "A colossal piece of machinery used to synthesize and distribute reagents throughout the station's plumbing. It stabilizes the reagents within, preventing reactions."
+		user << "A grimy dial stands at [output_pressure]."
+		user << "A dirt-smeared meter is counting at [synth_speed]."
+	else
+		user << "A huge piece of machinery. You aren't sure what it does."
+	if(synth_speed == max_synth_speed && output_pressure == max_output_pressure) //If it's all maxed out
+		user << "<span class='warning'><b>The metal creaks and moans with strain!</b></span>"
+
+/obj/machinery/plumbing/chemical_synthesizer/attack_hand(mob/user)
+	if(user.job != "Plumber")
```
Comment (RemieRichards) [inline code/game/machinery/plumbing/synthesizer.dm]: Restricting the use of the machine to the job? Bad play.
1. People cannot transfer to this job now via the HoP/whatever department assumes command of the Plumbers
2. It's just shit, ANYONE can fuck with atmos, so why not let ANYONE fuck with this?

Comment (Anonus) [inline code/game/machinery/plumbing/synthesizer.dm]: Don't do it, just make it super fucking complex.

**Comment by bgobandit** [comment]
:+1:

**Comment by Xhuis** [comment]
# Plumbing will not be restricted to plumbers, that has been scrapped

**Comment by Boredone** [comment]
So, whats the point of the heavy duty faucets?
I don't see them added yet from what I can glance at, but sinks last I checked already fill a container up by how much you have it set to transfer.

**Comment by daman997** [comment]
sounds cool i can dig it :+1:

**Comment by ike709** [comment]
plumbing.dmi tho

No really this is a cool idea and I love it :+1:

**Comment by PKPenguin321** [comment]
holy shit im so ready for this
mario hat traitor item for climbing in and out of sinks when

File: code/game/machinery/plumbing/synthesizer.dm (hunk 1 above)
Comment (Core0verload) [inline code/game/machinery/plumbing/synthesizer.dm]: Fuck no. I want to fill the station with banana juice as clown. Remove that restriction.

**Comment by WJohn** [comment]
I don't imagine we'll have actual water pipes around the station, will we?

I guess I'll offer sprites soon.

**Comment by fizbump** [comment]
Please add an emag option to replace all the water in the synthesizer with BEEEEEES

**Comment by WJohn** [comment]
Only when the bee PR gets merged.

Literally never.

File: code/game/machinery/plumbing/synthesizer.dm [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	var/max_synth_speed = 1000
+	var/min_synth_speed = 0
+
+/obj/machinery/plumbing/chemical_synthesizer/New()
+	..()
+	create_reagents(100000) //100k - a lot!
+	SSobj.processing |= src
+
+/obj/machinery/plumbing/chemical_synthesizer/Destroy()
+	SSobj.processing &= src
```
Comment (MrPerson) [inline code/game/machinery/plumbing/synthesizer.dm]: return ..()

File: code/game/machinery/plumbing/synthesizer.dm [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+/obj/machinery/plumbing/chemical_synthesizer/examine(mob/user)
+	..()
+	if(user.job == "Plumber")
+		user << "A colossal piece of machinery used to synthesize and distribute reagents throughout the station's plumbing. It stabilizes the reagents within, preventing reactions."
+		user << "A grimy dial stands at [output_pressure]."
+		user << "A dirt-smeared meter is counting at [synth_speed]."
+	else
+		user << "A huge piece of machinery. You aren't sure what it does."
+	if(synth_speed == max_synth_speed && output_pressure == max_output_pressure) //If it's all maxed out
+		user << "<span class='warning'><b>The metal creaks and moans with strain!</b></span>"
```
Comment (MrPerson) [inline code/game/machinery/plumbing/synthesizer.dm]: There was talk of dropping the hardcoded job restriction, but do people want to keep this more minor one as well? I'm just floating the question, not saying that we should or shouldn't.

Comment (Iamgoofball) [inline code/game/machinery/plumbing/synthesizer.dm]: I feel like this is a good example of how to properly do job bonuses.

A professional plumber who has been doing the job for space years should be able to know the ins and outs of the machine, while the clown should be confused.

**Comment by Jalleo** [comment]
Make this a part of cargo instead of having a new specific job. It could then have inbuilt upgrade system that can be ordered by cargo for them to upgrade it themselves.

This would at least add extra relevance to the job which has stagnated a bit.

File: code/game/machinery/plumbing/synthesizer.dm (hunk 3 above)
Comment (Jalleo) [inline code/game/machinery/plumbing/synthesizer.dm]: Maybe still have the pressure stated but in such a way you are not certain which is which.

Comment (GunHog) [inline code/game/machinery/plumbing/synthesizer.dm]: We do not have a skill system as Bay or similar codebases have. Enhanced examines based on the job the user holds is interesting, but should not result in complete restrictions.

File: code/game/machinery/plumbing/synthesizer.dm (hunk 1 above)
Comment (GunHog) [inline code/game/machinery/plumbing/synthesizer.dm]: Just do an access check, as everything else does. You have not even included the CE in this, and heads should always be able to do any job his subordinates can.

**Comment by Iamgoofball** [comment]
We want to add new jobs.

More jobs = more player slots = higher player cap.

**Comment by Core0verload** [comment]
One plumber is enought.

File: code/game/machinery/plumbing/synthesizer.dm (hunk 3 above)
Comment (MrPerson) [inline code/game/machinery/plumbing/synthesizer.dm]: If it's not a restriction, it's a fluff change. Which is fine, just be aware of that.

**Comment by MrPerson** [comment]
@Core0verload 
But then how will we have a red plumber and a green plumber?????

**Comment by CPTANT** [comment]
I feel this would be a lot more engaging if people actually had to drink.

poisoned water supply ftw

**Comment by Miauw62** [comment]
i dont really see the point of this, to  be honest.

**Comment by Ergovisavi** [comment]
If this introduces more piping I'm going to strangle someone.

**Comment by duncathan** [comment]
@WJohn @Ergovisavi can nobody read the OP
>Due to cutting-edge research, all objects that require reagents are automatically fed by miniature pipes embedded in the station's flooring itself. **This means that you will not have to lay down piping from place to place.**

**Comment by Xhuis** [comment]
MORE pipes would make mapping hell, so no more pipes.

File: code/game/machinery/plumbing/synthesizer.dm [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+/obj/machinery/plumbing/chemical_synthesizer/Destroy()
+	SSobj.processing &= src
+
+/obj/machinery/plumbing/chemical_synthesizer/process()
+	reagents.add_reagent(reagent_id, synth_speed) //Takes a decent amount of time to fill up
+
+/obj/machinery/plumbing/chemical_synthesizer/examine(mob/user)
+	..()
+	if(user.job == "Plumber")
+		user << "A colossal piece of machinery used to synthesize and distribute reagents throughout the station's plumbing. It stabilizes the reagents within, preventing reactions."
```
Comment (Thunder12345) [inline code/game/machinery/plumbing/synthesizer.dm]: >Prevents reactions within the machine
>Input nitro ingredients
>Kek at exploding sinks
Mwahahahahahaha

**Comment by vista-** [comment]
>no separate water pipes

such disappointment, very 2/10

**Comment by PKPenguin321** [comment]
I HAVE A DREAM
THAT ONE DAY, SOME BRAVE AMERICAN TRAITOR
REPLACES THE WATER SUPPLY WITH NANOMACHINES
AND SUBVERTS THE AI

File: code/game/machinery/plumbing/synthesizer.dm (hunk 4 above)
Comment (Iamgoofball) [inline code/game/machinery/plumbing/synthesizer.dm]: Feature

**Comment by kevinz000** [comment]
Will we be able to order a new one from cargo for a shitload of points if it's destroyed?

**Comment by Xhuis** [comment]
Yes, likely. I plan to have it react to most things in the same way as the gravgen though - repairable albeit nonfunctional.

**Comment by GunHog** [comment]
How will the sprinkler system work?

**Comment by Xhuis** [comment]
Whenever a fire alarm triggers in any given area, wall-mounted sprinklers come on and distribute the reagent they're filled with.

**Comment by freerealestate** [comment]
this might be pretty neat provided you don't fuck it up

one of the most original prs in years

**Comment by Xhuis** [comment]
Yeah, I know I said I'd do this. Yeah, yeah, yeah. I just don't have the time to do it. I'm going to close this PR, but i'm not going to delete the branch in case I do eventually decide to work on it again.

**Comment by PKPenguin321** [comment]
:cry:

**Pull request closed without merging**